content/doc/book/using/remote-access-api.adoc

```diff
@@ -0,0 +1,260 @@
+Remember that name in this command should match with File location in file parameter in job configuration.
+----
+
+image:../images/using/remote-access-api/file-parameter.png[image,title="File Parameter"]
+
+[[RemoteaccessAPI-RemoteAPIandsecurity]]
+== Remote API and security
+
+When your Jenkins is secured, you can use HTTP BASIC authentication to
+authenticate remote API requests. See
+https://wiki.jenkins.io/display/JENKINS/Authenticating+scripted+clients[Authenticating
+scripted clients] for more details.
+
+[[RemoteaccessAPI-CSRFProtection]]
+== CSRF Protection
+
+*Note*: From Jenkins 2.96 onward, you can use an API token and avoid
+using a crumb / CSRF token.
+
+If your Jenkins uses the "Prevent Cross Site Request Forgery exploits"
+security option (which it should), when you make a `+POST+` request, you
+have to send a CSRF protection token as an HTTP request header. +
+For curl/wget you can obtain the header needed in the request from the
+URL `+JENKINS_URL/crumbIssuer/api/xml+` (or `+.../api/json+`). Something
+like this:
+
+....
+wget -q --auth-no-challenge --user USERNAME --password PASSWORD --output-document - \
+'JENKINS_URL/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,":",//crumb)'
+....
+
+This will print something like "Jenkins-Crumb:1234abcd", which you
+should add to the subsequent request.
+
+[[RemoteaccessAPI-Samplecode]]
+== Sample code
+
+https://github.com/jenkinsci/extras-client-demo/tree/master/src/main/java/org/jvnet/hudson/client_demo/[A
+simple client is available] to demonstrate how you can invoke the XML
+from Java
+(https://github.com/jenkinsci/extras-client-demo/blob/master/src/main/java/org/jvnet/hudson/client_demo/Main.java[Java
+source])
+
+[[RemoteaccessAPI-XPathselection]]
+== XPath selection
+
+The XML API supports a selection by XPath by using the query parameter
+'xpath'. This is convenient for extracting information in environments
+where XML manipulation is tedious (such as shell script.) See
+https://issues.jenkins-ci.org/browse/JENKINS-626[issue #626] for an
+example of how to use this. +
+See `+.../api/+` on your Jenkins server for more up-to-date details.
+
+[[RemoteaccessAPI-XPathexclusion]]
+=== XPath exclusion
+
+Similar to the 'xpath' query parameter above, you can use (possibly
+multiple) 'exclude' query patterns to exclude nodes from the resulting
+XML. All the nodes that match the specified XPath will be removed from
+the XML. +
+See `+.../api/+` on your Jenkins server for more up-to-date details.
+
+[[RemoteaccessAPI-Depthcontrol]]
+== Depth control
+
+Sometimes the remote API doesn't give you enough information in one
+call. For example, if you'd like to find out all the last successful
+build of a given view, you'd realize that the invocation to the remote
+API of the view won't give you this, and you'd have to recursively call
+the remote API of each project to find this out. The depth control,
+introduced in 1.167, solves this problem. To understand this feature,
```

Comment:
No benefit to referencing a Jenkins version from 5+ years ago.  We can safely act like the capability has always been available.
```suggestion
solves this problem. To understand this feature,
```